parser.go

```diff
@@ -135,28 +135,36 @@ func (p *Parser) ParseUnverified(tokenString string, claims Claims) (token *Toke
 		}
 		return token, parts, newError("could not base64 decode header", ErrTokenMalformed, err)
 	}
-	if err = json.Unmarshal(headerBytes, &token.Header); err != nil {
+	if err := json.Unmarshal(headerBytes, &token.Header); err != nil {
```

Comment:
any specific reasons for re-defining / shadowing `err` here?

Reply:
That is a good question that sent me looking for answers... but I found no official guidance. It does get a bit tricky especially with named return values, but this function has no empty `return`. I like it because it contains the scope of the error to that specific block, but, yeah, it could get confusing with the named returned value. I'll revert it. Thanks!